### PR TITLE
Bubble up workflows errors

### PR DESCRIPTION
**Pull request by colinking**
This PR serializes `HTTPError` errors s.t. they are rendered as error output in the parent run.

The approach taken in this PR allows a client (e.g. a workflow) to access the original error (e.g. from the underlying activity execution) as if the error had been thrown within the same process. This requires custom JSON (de)serialization logic, so we'll need to implement this for other errors we add.

Before:

CleanShot 2022-09-28 at 14 01 39

After:

CleanShot 2022-09-28 at 13 58 21

cc @calufornia 

Resolves TSK-452

### CODE REVIEW

**Comment by colinking** [comment]
Current dependencies on/for this PR:
* main
  * **PR #152** <a href="https://app.graphite.dev/github/pr/airplanedev/node-sdk/152" target="_blank">Graphite</a> 
    * **PR #153** <a href="https://app.graphite.dev/github/pr/airplanedev/node-sdk/153" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/airplanedev/node-sdk/153?utm_source=stack-comment).

File: packages/workflow-runtime/internal/index.ts [hunk 1]
```diff
@@ -13,72 +12,12 @@ import {
...
-const { executeTaskActivity } = proxyActivities<ReturnType<typeof createActivities>>({
-  startToCloseTimeout: "120s",
-  retry: {
-    initialInterval: "3s",
-    backoffCoefficient: 2,
-    maximumAttempts: 5,
-  },
-});
+// Exported so that it can be imported by the worker.
+export { createActivities } from "./activities";
```
Comment (colinking) [inline packages/workflow-runtime/internal/index.ts:20]: I extracted the activities to their own file so I could wrap them with (de)serialization logic.

File: packages/airplane/internal/api/fetcher.ts [hunk 2]
```diff
@@ -154,38 +156,85 @@ export class Fetcher {
...
+  public readonly errorCode?: string;
+
+  constructor(
+    message: string,
+    options: {
+      method: string;
+      url: string;
+      statusCode: number;
+      errorCode?: string;
+    }
```
Comment (colinking) [inline packages/airplane/internal/api/fetcher.ts:176]: I added some extra metadata to these errors to make HTTP errors easier to debug when marshaled.

**Review by calufornia — APPROVED** [review]
lgtm! My only suggestion is to maybe move the wrapper functions (e.g. `getRunOutput`, `createPrompt` etc.) outside of `activities.ts` since I guess they technically aren't activities, but no strong opinions there.

**Review by ericliuche — APPROVED** [review]

**Pull request merged**